javalib/src/main/scala/java/util/Arrays.scala

```diff
@@ -815,13 +814,13 @@ object Arrays {
   }
 
   @inline
-  private def checkRangeIndices(length: Int, start: Int, end: Int): Unit = {
+  private def checkRangeIndices(a: Array[_], start: Int, end: Int): Unit = {
     if (start > end)
       throw new IllegalArgumentException("fromIndex(" + start + ") > toIndex(" + end + ")")
-    SemanticsUtils.arrayIndexOutOfBoundsCheck(start < 0,
-        new ArrayIndexOutOfBoundsException("Array index out of range: " + start))
-    SemanticsUtils.arrayIndexOutOfBoundsCheck(end > length,
-        new ArrayIndexOutOfBoundsException("Array index out of range: " + end))
+
+    // bounds checks
+    a(start)
+    a(end)
```

Comment:
This does not do the same thing as before if `end == a.length` (including when `start == end == a.length`); it now throws. I doubt this is correct.